data-processing/entities/citations/commands/upload_citations.py

```diff
@@ -34,9 +36,60 @@ def get_description() -> str:
     def get_arxiv_ids_dirkey(self) -> str:
         return "citations-locations"
 
+    def _get_bibitem_texts_from_bibitems(
+        self,
+        arxiv_id: ArxivId,
+        bibitems: List[Bibitem]
+    ) -> Dict[str, str]:
+        def acceptable_str(maybe_str: Optional[str]) -> bool:
+            # Not perfect but should weed some stuff out.
+            return maybe_str is not None and maybe_str.strip() != ""
+
+        bibitem_texts: Dict[str, str] = {}
+        for bibitem in bibitems:
+            maybe_bibitem_id = bibitem.id_
+            maybe_bibitem_text = bibitem.text
+            if acceptable_str(maybe_bibitem_id) and acceptable_str(maybe_bibitem_text):
+                if maybe_bibitem_id in bibitem_texts:
```

Comment:
a little hard to follow; can give an example in comment what this is tryin to solve?

Reply:
Will do!